cvmfs/options.cc

```diff
@@ -71,6 +71,23 @@ string OptionsManager::TrimParameter(const string &parameter) {
   return result;
 }
 
+void OptionsManager::SwitchTemplateManager(
+  OptionsTemplateManager *opt_templ_mgr_param) {
+  delete opt_templ_mgr_;
+  if (opt_templ_mgr_param != NULL) {
+    opt_templ_mgr_ = opt_templ_mgr_param;
+  } else {
+    opt_templ_mgr_ = new OptionsTemplateManager();
+  }
+  for (std::map<std::string, std::string>::iterator it
+    = templatable_values_.begin();
+    it != templatable_values_.end();
+    it++) {
+    config_[it->first].value = it->second;
+    opt_templ_mgr_->ParseString(&(config_[it->first].value));
```

Comment:
Shouldn't this be a call to `PopulateParameter` to change the environment as well?

Reply:
I have just submitted a commit including the requested changes. Concerning this change request I believe it could be better to have an extra method which changes the parameter based environment variables. The disadvantage of using `PopulateParameter` would be that a protected parameters can no longer be updated based on template variables.

For now I have implemented the method I proposed above, but I can also change this in the way described you described...